### ISSUE TITLE
Modify the number of devices in the proxy/ gateway levels

### ISSUE THREAD

**Issue by imene-gcm** [sank20144/iFogSimDec2021#1]
Hi,

I am actually creating a topology using proxy servers and gateways. As described in the reference article, the default number of gateways is 118 and the number of proxies is fixed to 12.

- Is it possible to modify the number of these devices? If yes, how to do that please?

Thank you in advance!
Imene.

**Comment by sank20144** [comment]
Yes you can modify them. It just have to be added to the application loop. If you check Random_placement.java you can easily see how it can be done

**Comment by imene-gcm** [comment]
Hi! I really appreciate your reply. However, there is no class called  Random_placement.java / RandomPlacement.java. 
Could you please give me more details?

Thank you!
Imene.

**Comment by sank20144** [comment]
https://github.com/sank20144/iFogSimDec2021/blob/main/src/org/fog/test/perfeval/Random_Placement.java

**Comment by imene-gcm** [comment]
Thank you a lot ! I understand what you mean. 

But, I want to have a very huge number of nodes, I don't want to create them one by one. A "for loop" is not working since I have to use a locator object to collect their positons in order to migrate their modules. 


So, I've created a level named FOG, but it is set to 12 by default. I'm searching where the default properties are described.


` for (int i = 0; i < locator.getLevelWiseResources(locator.getLevelID("Fog")).size(); i++) {

             FogDevice fog = createFogDevice("fog_" + i, 2800, 4000, 10000, 10000, 0.0, 107.339, 83.4333); // creates the fog device (level=1)
             locator.linkDataWithInstance(fog.getId(), locator.getLevelWiseResources(locator.getLevelID("Fog")).get(i));
             fog.setParentId(cloud.getId()); 
             fog.setUplinkLatency(100); // latency of connection 
             fogDevices.add(fog);

         }`





Thank you!

**Comment by sank20144** [comment]
Sorry I'm not sure about what you're trying to do. If you explain it a bit more then I might be able to help
